src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2934,7 +2934,8 @@ public IEnumerable<ISymbol> GetSymbolsWithName(Func<string, bool> predicate, Sym
                 var result = new HashSet<ISymbol>();
                 var spine = new List<MergedNamespaceOrTypeDeclaration>();
 
-                AppendSymbolsWithName(spine, _compilation.Declarations.MergedRoot, predicate, filter, result, cancellationToken);
+                var mergedRoot = ((SourceNamespaceSymbol)_compilation.SourceModule.GlobalNamespace).MergedDeclaration;
```

Comment:
Should we be concerned with the fact that now we are always creating SourceModuleSymbol here, which means that references are getting resolved etc., whereas before, if result is empty, it looks like we could have avoided doing this?